check_and_format.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+set -e
+
+pushd headers
```

Comment:
If one of the `make` commands fails, I think the script will exit before the `popd`, and the working directory won't be reset. Can we use `make -C headers` instead of `make`? This tells `make` to change directories before running its rules, without needing to change directories manually.

Reply:
That’s not a problem if the directory is invalid, given the script is called as a child process, and so change of the current working dir by the child won’t impact the parent process (the user’s shell).

Reply:
Oh yeah you're right. Never mind then.

Reply:
Actually, now that I think about this more, `pushd` and `popd` are bash-only (`/bin/bash`), not POSIX shell (`/bin/sh`). So I would still prefer `make -C headers` since that's more compatible if someone's default shell isn't bash.

Reply:
Thanks for the info. Will do.
(actually, you already did that)